application-analytics-ui/src/main/resources/Analytics/Code/Macros/VisitsSummary.xml

```diff
@@ -0,0 +1,277 @@
+        <restricted>0</restricted>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require.config({
+  paths: {
+    chart: "$services.webjars.url('org.webjars:chartjs', 'Chart.bundle.min.js')"
+  }
+});
+
+require(['jquery', 'chart'], function ($) {
+  const SELECTORS = {
+    options: '.analytics_genericVisitsFields',
+    parent: '.analytics_graphMacro',
+    canvas: 'canvas',
+    macroOptions: '.analytics_visitsGenericMacroOptions',
+    period: '.analytics_visitsGenericPeriods',
+    updateCanvas: '.analytics_visitsGenericPeriods, .analytics_genericVisitsFields'
+  };
+  const RGB = [`rgb(255, 0, 0)`, `rgb(0, 255, 0)`, `rgb(0, 0, 255)`]
+
+  // A map to store charts for dynamic updates.
+  let chartsMap = new Map();
+
+  // Listen for changes on the period or the field that is displayed and update the chart.
+  $(document).on('change', SELECTORS.updateCanvas, function () {
+    let parent = $(this).closest(SELECTORS.parent);
+    let cacheId = parent.attr('id');
+    let timeInterval = parent.find(SELECTORS.macroOptions).data('date');
+    let canvas = parent.find(SELECTORS.canvas);
+    let yAxeNames = $('option:selected', parent.find(SELECTORS.options)).data('label').split('&amp;');
+    let endpoints = $('option:selected', parent.find(SELECTORS.options)).data('endpoints').split('&amp;');
+    let fields =parent.find(SELECTORS.options).val().split('&amp;');
```

Comment:
I'm not sure of the format for these string, but it looks like they are query string? You should check https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams , since splitting it's fragile when you are not sure of the values
Also, you have a space missing

Reply:
These are not query parameters but endpoints that I need to call to display a multi-line graph, each endpoint representing a line. Also, I'm certain that the endpoints would never contain the '&' character.